### PR TITLE
AAP-33634: ModelPipelines: Health endpoint improvements

### PR DESCRIPTION
**Pull request by manstis**
Jira Issue: https://issues.redhat.com/browse/AAP-33634

## Description
Add/expose "Health Check" for each pipeline.

The "aliases" of "Completions" and "Chat Bot" have been preserved as `model-server` and `chatbot-service` respectfully.

This means we _should not_ have to update our test suites - nor anybody else who depends on the existing names.

## Testing
Run the service locally and check the _health_ endpoint: http://127.0.0.1:8000/check/status/

Example output:
```
{
  "timestamp": "2024-12-06T09:38:32.625598",
  "version": "image-tags-not-defined",
  "git_commit": "git-commit-hash-not-defined",
  "deployed_region": "unknown",
  "status": "error",
  "dependencies": [
    {
      "name": "secret-manager",
      "status": "ok",
      "time_taken": 466.697
    },
    {
      "name": "authorization",
      "status": "ok",
      "time_taken": 0.032
    },
    {
      "name": "db",
      "status": "ok",
      "time_taken": 39.233
    },
    {
      "name": "chatbot-service",
      "status": {
        "provider": "http",
        "models": "unavailable: An error occurred" ## <!-- [manstis] I don't have a public route to the ChatBot backend
      },
      "time_taken": 41.034
    },
    {
      "name": "model-server",
      "status": {
        "provider": "wca",
        "tokens": "ok",
        "models": "ok"
      },
      "time_taken": 3394.077
    },
    {
      "name": "content-match",
      "status": "disabled",
      "time_taken": 0.002
    },
    {
      "name": "playbook-explanation",
      "status": "disabled",
      "time_taken": 0.002
    },
    {
      "name": "playbook-generation",
      "status": "disabled",
      "time_taken": 0.001
    },
    {
      "name": "role-generation",
      "status": "disabled",
      "time_taken": 0.002
    }
  ]
}
```

### Steps to test
1. Pull down the PR
2. Run the service locally
3. Check the _health_ endpoint http://127.0.0.1:8000/check/status/

### Scenarios tested
As above.

## Production deployment
- [x] This code change is ready for production on its own
- [ ] This code change requires the following considerations before going to production:

### CODE REVIEW

**Comment by manstis** [comment]
@TamiTakamiya @goneri @jameswnl For your consideration.

File: ansible_ai_connect/healthcheck/apps.py [hunk 1]
```diff
@@ -20,14 +20,17 @@ class HealthCheckAppConfig(AppConfig):
...
-            ChatbotServiceHealthCheck,
-            ModelServerHealthCheck,
+            ModelPipelineHealthCheck,
         )
 
-        plugin_dir.register(ModelServerHealthCheck)
         plugin_dir.register(AWSSecretManagerHealthCheck)
         plugin_dir.register(AuthorizationHealthCheck)
-        plugin_dir.register(ChatbotServiceHealthCheck)
+        for pipeline in REGISTRY_ENTRY.keys():
```
Comment (manstis) [inline ansible_ai_connect/healthcheck/apps.py:34]: This is the _meat_ of the PR.

I register each _pipeline_ with Django's Health Service.

File: ansible_ai_connect/healthcheck/backends.py [hunk 2]
```diff
@@ -76,34 +76,6 @@ def pretty_status(self):
         return str(super().pretty_status())
 
 
-class ModelServerHealthCheck(BaseLightspeedHealthCheck):
```
Comment (manstis) [inline ansible_ai_connect/healthcheck/backends.py:79]: No longer needed.

File: ansible_ai_connect/healthcheck/backends.py [hunk 3]
```diff
@@ -141,32 +113,37 @@ def identifier(self):
         return self.__class__.__name__
 
 
-class ChatbotServiceHealthCheck(BaseLightspeedHealthCheck):
```
Comment (manstis) [inline ansible_ai_connect/healthcheck/backends.py:144]: The ChatBot health check has [moved](https://github.com/ansible/ansible-ai-connect-service/pull/1431/commits/4628048859cb69e1952b978b10f33cfeb2d20c73#diff-b6c900f30e6213219c6ddcd3af4d9405581fee6029a05e493bdd4d9613268578R171-R198) to it's _pipeline_.

File: ansible_ai_connect/healthcheck/backends.py [hunk 4]
```diff
@@ -141,32 +113,37 @@ def identifier(self):
...
+class ModelPipelineHealthCheck(BaseLightspeedHealthCheck):
```
Comment (manstis) [inline ansible_ai_connect/healthcheck/backends.py:116]: This is a generic wrapper around the different _pipelines_.

File: ansible_ai_connect/healthcheck/views.py [hunk 5]
```diff
@@ -55,12 +55,14 @@ def common_data():
 class HealthCheckCustomView(MainView):
     _plugin_name_map = {
         "DatabaseBackend": "db",
-        "ModelServerHealthCheck": "model-server",
         "AWSSecretManagerHealthCheck": "secret-manager",
-        "WCAHealthCheck": "wca",
-        "WCAOnPremHealthCheck": "wca-onprem",
         "AuthorizationHealthCheck": "authorization",
-        "ChatbotServiceHealthCheck": "chatbot-service",
+        "ModelPipelineCompletions": "completions-service",
```
Comment (manstis) [inline ansible_ai_connect/healthcheck/views.py:60]: I'd like to move the alias to the pipeline itself and make this block more _dynamic_.

**Review by TamiTakamiya — DISMISSED** [review]
LGTM. I was thinking that this PR was for healthcheck, but it was more on pipelines :-)

**Review by TamiTakamiya — APPROVED** [review]

**Pull request merged**